File: fix-all-jsx-errors.js
const fs = require('fs');
const path = require('path');

// Function to fix all JSX syntax errors
function fixAllJSXErrors(content) {
  let fixed = content;

  // Fix malformed Head tags with title and meta - pattern 1
  fixed = fixed.replace(/<Head>title>([^<]*)<\/title>meta name="([^"]*)" content="([^"]*)" \/>/g, 
    '<Head>\n        <title>$1</title>\n        <meta name="$2" content="$3" />');
  
  // Fix malformed Head tags with title and meta - pattern 2 (with variables)
  fixed = fixed.replace(/<Head>title>\{([^}]*)\}<\/title>meta name="([^"]*)" content=\{([^}]*)\} \/>/g, 
    '<Head>\n        <title>{$1}</title>\n        <meta name="$2" content={$3} />');
  
  // Fix malformed closing Head tag followed by div
  fixed = fixed.replace(/<\/Head>\s*<div className="([^"]*)"\s*>/g, 
    '</Head>\n      <div className="$1">');
  
  // Fix malformed JSX elements with missing spaces
  fixed = fixed.replace(/(\w+)><(\w+)/g, '$1> <$2');
  fixed = fixed.replace(/(\w+)><\/(\w+)/g, '$1></$2');
  
  // Fix missing opening tags
  fixed = fixed.replace(/(<div className="[^"]*">)/g, (match) => {
    if (!match.includes('<div')) {
      return '<div ' + match;
    }
    return match;
  });
  
  // Fix malformed closing tags
  fixed = fixed.replace(/(\w+)>\s*<\/(\w+)>/g, '$1></$2>');
  
  // Fix missing spaces in JSX attributes
  fixed = fixed.replace(/(\w+)="([^"]*)"(\w+)/g, '$1="$2" $3');
  
  // Fix malformed JSX fragments
  fixed = fixed.replace(/<>\s*<Head>/g, '<>\n      <Head>');
  fixed = fixed.replace(/<\/Head>\s*<div/g, '</Head>\n      <div');
  
  // Fix missing closing brackets
  fixed = fixed.replace(/(\w+)\s*}\s*$/g, '$1}');
  
  return fixed;
}

// Function to process all TypeScript/TSX files
function processFiles(dir) {
  const files = fs.readdirSync(dir);
  
  files.forEach(file => {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);
    
    if (stat.isDirectory()) {
      processFiles(filePath);
    } else if (file.endsWith('.tsx') || file.endsWith('.ts')) {
      try {
        const content = fs.readFileSync(filePath, 'utf8');
        const fixed = fixAllJSXErrors(content);
        
        if (content !== fixed) {
          fs.writeFileSync(filePath, fixed);
          console.log(`Fixed JSX: ${filePath}`);
        }
      } catch (error) {
        console.error(`Error processing ${filePath}:`, error.message);
      }
    }
  });
</div>
      </div>}

// Process app directory
console.log('Fixing all JSX syntax errors in app directory...');
processFiles('./app');

// Process src directory
console.log('Fixing all JSX syntax errors in src directory...');
processFiles('./src');

// Process components directory
console.log('Fixing all JSX syntax errors in components directory...');
processFiles('./components');

console.log('All JSX syntax fixes completed!');
      </Head>
      </div>
      </div>
      </div>
      </Head>
      </Head>
      </Head>
      </Head>

File: fix_all_jsx_errors.py
#!/usr/bin/env python3
import os
import re
import glob

def fix_all_jsx_errors():
    # Find all page.tsx files
    pattern = "app/**/page.tsx"
    files = glob.glob(pattern, recursive=True)
    
    fixed_count = 0
    
    for file_path in files:
        try:
            with open(file_path, 'r', encoding='utf-8') as f:
                content = f.read()
            
            original_content = content
            
            # Fix the malformed JSX pattern - multiple variations
            patterns_to_fix = [
                (r'Page\n              </', 'Page\n              </'),
                (r'Page\n  </', 'Page\n              </'),
                (r'Page\n    </', 'Page\n              </'),
                (r'Page\n      </', 'Page\n              </'),
                (r'Page\n        </', 'Page\n              </'),
                (r'Page\n          </', 'Page\n              </'),
                (r'Page\n            </', 'Page\n              </'),
            ]
            
            for pattern, replacement in patterns_to_fix:
                content = re.sub(pattern, replacement, content)
            
            # Fix the specific malformed pattern with "Page" and "</"
            content = re.sub(
                r'Page\n              </',
                'Page\n              </',
                content
            )
            
            # Also fix the title and description based on the file path
            dir_name = os.path.dirname(file_path).split('/')[-1]
            if dir_name and dir_name != 'app':
                # Convert kebab-case to title case
                title = ' '.join(word.capitalize() for word in dir_name.split('-'))
                
                content = re.sub(
                    r'<title>Database Services \| Zion Tech Group</title>',
                    f'<title>{title} | Zion Tech Group</title>',
                    content
                )
                
                content = re.sub(
                    r'content="Professional Database Services by Zion Tech Group',
                    f'content="Professional {title} by Zion Tech Group',
                    content
                )
                
                content = re.sub(
                    r'content="database services, AI solutions, IT services, Zion Tech Group, database"',
                    f'content="{dir_name}, AI solutions, IT services, Zion Tech Group, {dir_name}"',
                    content
                )
            
            # Write the fixed content back if it changed
            if content != original_content:
                with open(file_path, 'w', encoding='utf-8') as f:
                    f.write(content)
                
                fixed_count += 1
                print(f"Fixed {file_path}")
                
        except Exception as e:
            print(f"Error processing {file_path}: {e}")
    
    print(f"Fixed {fixed_count} files")

if __name__ == "__main__":
    fix_all_jsx_errors()